packages/frontend/src/components/swap/components/TransactionDetails.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+
+import classNames from '../../../utils/classNames';
+import Accordion from '../../common/Accordion';
+import AccordionTrigger from '../../send/components/AccordionTrigger';
+import Breakdown from '../../send/components/css/Breakdown.css';
+import Amount from '../../send/components/entry_types/Amount';
+import { exchangeRateTranslation } from './helpers';
+import SlippagePicker from './SlippagePicker';
+
+const TransactionDetails = ({
+    selectedTokenFrom,
+    selectedTokenTo,
+    estimatedFeesInNear,
+    estimatedMinReceived,
+    amount,
+    exchangeRate,
+    tradingFee,
+    setSlippage
+}) => {
+    const [open, setOpen] = useState(false);
+
+    const commissionFee = tradingFee?.toFixed(5);
+    const isUSNSwap = selectedTokenFrom.onChainFTMetadata.name === 'USN' ||
+        selectedTokenTo.onChainFTMetadata.name === 'USN';
+
+    const minimumReceived = exchangeRateTranslation({
+        inputtedAmountOfToken: selectedTokenFrom,
+        calculateAmountOfToken: selectedTokenTo,
+        balance: amount,
+        exchangeRate
+    }) - tradingFee;
+
+    return (
+        <Breakdown
+            className={classNames([
+                'transaction-details-breakdown',
+                open ? 'open' : '',
+            ])}
+        >
+            <Accordion
+                trigger="transaction-details-breakdown"
+                className="breakdown"
+            >
+                {isUSNSwap && 
+                    <SlippagePicker
+                        translateIdTitle={'swap.priceImpact'}
+                        translateIdInfoTooltip="swap.translateIdInfoTooltip.priceImpact"
+                        setSlippage={setSlippage}
+                    />
+                }
+                
+                {!isUSNSwap && 
```

Comment:
What do you think about making separate `TransactionsDetails` components for USN and wNEAR swapping to make it easier to read and avoid all the conditionals?

Reply:
yeah that makes sense, will do 👍